File: 2630-alternating-digit-sum/2630-alternating-digit-sum.py
class Solution:
    def alternateDigitSum(self, n: int) -> int:
        s=(str(n))[::-1]
        j=int(s)
        i=0
        k=0
        while(j):
            rem=j%10
            if(i%2!=0):
                k-=(rem)
            else:
                k+=rem
            j//=10
            i+=1
        return k


        

File: 2630-alternating-digit-sum/2630-alternating-digit-sum.cpp
class Solution {
public:
    int alternateDigitSum(int n) {
        int k=0;
        while(n){
            int rem=n%10;
            k=k*10+rem;
            n/=10;
        }
        int i=0;
        int sum=0;
        while(k){
            int rem=k%10;
            if(i%2!=0){
                sum-=rem;
            }
            else{
                sum+=rem;
            }
            k/=10;
            i++;
        }
        return sum;
        
    }
};